packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/redshift-cluster.ts

```diff
@@ -0,0 +1,276 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as firehose from '@aws-cdk/aws-kinesisfirehose';
+import * as kms from '@aws-cdk/aws-kms';
+import * as redshift from '@aws-cdk/aws-redshift';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { BackupMode, CommonDestinationProps, CommonDestinationS3Props, Compression } from './common';
+import { createBackupConfig, createBufferingHints, createEncryptionConfig, createLoggingOptions, createProcessingConfig } from './private/helpers';
+import { FirehoseRedshiftTable } from './redshift/table';
+import { FirehoseRedshiftUser } from './redshift/user';
+
+/**
+ * The Redshift user Firehose will assume to deliver data to Redshift
+ */
+export interface RedshiftUser {
+  /**
+   * Username for user that has permission to insert records into a Redshift table.
+   */
+  readonly username: string;
+
+  /**
+   * Password for user that has permission to insert records into a Redshift table.
+   *
+   * Do not put passwords in your CDK code directly.
+   *
+   * @default - a Secrets Manager generated password.
+   */
+  readonly password?: cdk.SecretValue;
+
+  /**
+   * KMS key to encrypt the generated secret.
```

Comment:
Is this referring to the secret that will be generated to store the Redshift credentials (username and password)?

Should we also allow customers to just pass in a SecretsManager secret that already contains their Redshift credentials?